runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -272,165 +547,65 @@ protected ILogData read(final long address, long readStartTime) {
-        } else {
-            // Select everything in the read queue between
-            // the start and maxGlobal
-            readSet.addAll(context.readQueue.headSet(maxGlobal, true));
-        }
-        List<Long> toRead = readSet.stream()
-                .collect(Collectors.toList());
-
-        // The list to store read results in
-        List<ILogData> readFrom = readAll(toRead).stream()
-                .filter(x -> x.getType() == DataType.DATA)
-                .filter(x -> x.containsStream(context.id))
-                .collect(Collectors.toList());
-
-        // If any entries change the context,
-        // don't return anything greater than
-        // that entry
-        Optional<ILogData> contextEntry = readFrom.stream()
-                .filter(contextCheckFn::apply).findFirst();
-        if (contextEntry.isPresent()) {
-            log.trace("getNextEntries[{}] context switch @ {}", this,
-                    contextEntry.get().getGlobalAddress());
-            int idx = readFrom.indexOf(contextEntry.get());
-            readFrom = readFrom.subList(0, idx + 1);
-            // NOTE: readSet's clear() changed underlying context.readQueue
-            readSet.headSet(contextEntry.get().getGlobalAddress(), true).clear();
-        } else {
-            // Clear the entries which were read
-            context.readQueue.headSet(maxGlobal, true).clear();
-        }
-
-        // Transfer the addresses of the read entries to the resolved queue
-        readFrom.stream()
-                .forEach(x -> addToResolvedQueue(context, x.getGlobalAddress(), x));
-
-        // Update the global pointer
-        if (readFrom.size() > 0) {
-            context.setGlobalPointerCheckGCTrimMark(readFrom.get(readFrom.size() - 1)
-                    .getGlobalAddress());
-        }
-
-        return readFrom;
-    }
-
     /**
-     * Fill the read queue for the current context. This method is called
+     * Fill the read queue for the current stream. This method is called
      * whenever a client requests a read, but there are no addresses left in
      * the read queue.
      *
-     * <p>This method returns true if entries were added to the read queue,
-     * false otherwise.
-     *
      * @param maxGlobal     The maximum global address to read to.
-     * @param context       The current stream context.
-     *
      * @return              True, if entries were added to the read queue,
      *                      False, otherwise.
-     *
-     * {@inheritDoc}
      */
-    protected boolean fillReadQueue(final long maxGlobal,
-                                    final QueuedStreamContext context) {
-        log.trace("Fill_Read_Queue[{}] Max: {}, Current: {}, Resolved: {} - {}", this,
-                maxGlobal, context.getGlobalPointer(), context.maxResolution, context.minResolution);
-        log.trace("Fill_Read_Queue[{}]: addresses in this stream Resolved queue {}" +
+    public boolean fillReadQueue(final long maxGlobal) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'fillReadQueue(long)' has an NPath complexity of 1944](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651476)